tests/test_storm_control.py

```diff
@@ -0,0 +1,528 @@
+            if fv[0] == "SAI_POLICER_ATTR_CIR":
+                bps = fv[1]
+
+        #Retrieved value of bps from ASIC_DB is converted back to user input kbps
+        kbps = int(bps) / int(1000) * 8
+        print ("KBPS value")
+        print (kbps)
+
+        assert str(kbps) == str(kbps_value)
+
+
+    def add_storm_control_on_interface(self,dvs,if_name,storm_type,kbps_value):
+        print ("interface")
+        print (if_name)
+        print ("storm_type")
+        print (storm_type)
+        print ("kbps_value")
+        print (kbps_value)
+        self.add_storm_session(if_name, storm_type, kbps_value)
+        self.check_storm_control_on_interface(dvs,if_name,storm_type,kbps_value)
+
+    def test_add_storm_all_interfaces(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        tbl = swsscommon.Table(self.cdb,"PORT")
+        for key in tbl.getKeys():
+            self.add_storm_control_on_interface(dvs,key,"broadcast",1000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-unicast",2000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-multicast",3000000)
+
+    def test_warm_restart_all_interfaces(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        tbl = swsscommon.Table(self.cdb,"PORT")
+        for key in tbl.getKeys():
+            self.add_storm_control_on_interface(dvs,key,"broadcast",1000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-unicast",2000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-multicast",3000000)
+        dvs.runcmd("config save -y")
+        # enable warm restart
+        (exitcode, result) = dvs.runcmd("config warm_restart enable swss")
+        assert exitcode == 0
+
+        # freeze orchagent for warm restart
+        (exitcode, result) = dvs.runcmd("/usr/bin/orchagent_restart_check")
+        assert result == "RESTARTCHECK succeeded\n"
+        time.sleep(2)
+
+        dvs.stop_swss()
+        time.sleep(10)
+        dvs.start_swss()
+        time.sleep(10)
+
+        for key in tbl.getKeys():
+            self.check_storm_control_on_interface(dvs,key,"broadcast",1000000)
+            self.check_storm_control_on_interface(dvs,key,"unknown-unicast",2000000)
+            self.check_storm_control_on_interface(dvs,key,"unknown-multicast",3000000)
+
+    def test_add_storm_lag_interface(self,dvs,testlog):
+        self.setup_db(dvs)
+        lag_name = "PortChannel10"
+        member_interface = "Ethernet0"
+        kbps_value = 1000000
+        storm_list = ["broadcast","unknown-unicast","unknown-multicast"]
+        kbps_value_list = [1000000,2000000,3000000]
+
+        #Create VLAN interface and add member
+        self.create_port_channel(dvs,lag_name)
+        self.add_port_channel_member(dvs,lag_name,member_interface)
+
+        #click CLI verification
+        for storm_type in storm_list:
+            dvs.runcmd("config interface storm-control "+storm_type+" add "+lag_name+" "+str(kbps_value))
```

Comment:
Why isn't validation not done at the CLI level? do not allow config on Vlan/PortChannel interface

Reply:
Validation is done in CLI level already. This was written to validate Orchagent behavior.

config interface storm-control broadcast add Vlan200 1000
add broadcast storm-control
Storm-control is supported only on Ethernet interfaces. Not supported on Vlan200
Usage: config interface storm-control broadcast add [OPTIONS] <port_name>
                                                    <kbps_value>

Error: Unable to add broadcast storm-control